### PR TITLE
UIOR-432 Add or update call number

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
Add or update call number during receiving/checkin
https://issues.folio.org/browse/UIOR-432
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/CheckIn/ItemsListModal.js [hunk 1]
```diff
@@ -49,6 +61,15 @@ const ItemsListModal = ({
         marginBottom0
       />
     ),
+    'callNumber': (item) => (
+      <TextField
+        disabled={isLoading || item.pieceFormat !== 'Physical'}
```
Comment (aliaksei-chumakou) [inline src/components/CheckIn/ItemsListModal.js]: maybe it's better to use constant for 'Physical'

File: src/components/CheckIn/ItemsListModal.js [hunk 2]
```diff
@@ -49,6 +61,15 @@ const ItemsListModal = ({
...
+        onChange={(e) => onChangeField(item, e.target.value, 'callNumber')}
```
Comment (aliaksei-chumakou) [inline src/components/CheckIn/ItemsListModal.js]: maybe it's better to use `useCallback` for this

File: src/components/Receiving/LineDetails.js [hunk 3]
```diff
@@ -48,6 +59,16 @@ const LineDetails = ({
     'format': (item) => (
       <KeyValue value={get(item, 'pieceFormat', '')} />
     ),
+    'callNumber': (item) => (
+      <TextField
+        data-test-call-number
+        disabled={isLoading || itemsMap[item.itemId] === undefined}
+        onChange={(e) => onChangeField(item, e.target.value, 'callNumber')}
```
Comment (aliaksei-chumakou) [inline src/components/Receiving/LineDetails.js]: Maybe it's a good idea for `useCallback` here

File: src/components/CheckIn/ItemsListModal.js (hunk 2 above)
Comment (NikitaSedyx) [reply src/components/CheckIn/ItemsListModal.js]: I think there is no way to do this here as it depends on item

Comment (aliaksei-chumakou) [reply src/components/CheckIn/ItemsListModal.js]: if you look into `onChangeField = (item, value, key) => {` there is no `item` usage - only item.id. And this also could be passed through MCL - `this.props.onRowClick(e, row);`

Comment (aliaksei-chumakou) [reply src/components/CheckIn/ItemsListModal.js]: due to next tasks for separate receiving app maybe it's better to provide as less changes as possible

**Pull request merged**